grid2op/Reward/_alertTrustScore.py

```diff
@@ -0,0 +1,163 @@
+
+        # TODO
+        #lines_alerted_beforeattack = np.equal(env._time_since_last_alert, env._time_since_last_attack + 1) and lines_attacked
+        #self.total_nb_alerts += np.sum(lines_alerted_beforeattack)
+            
+        if not is_done:
+
+            # TODO
+            #lines_attacked_no_blackout = env._time_since_last_attack == SURVIVOR_TIMESTEPS
+#
+            #self.alert_attack_no_blackout += np.sum(lines_alerted_beforeattack[lines_attacked_no_blackout])
+            #self.no_alert_attack_no_blackout += np.sum(~lines_alerted_beforeattack[lines_attacked_no_blackout])
+            
+            return score_ep
+            
+        else:
+            self.nb_last_attacks=np.sum(np.any(self._ts_attack,axis=1))
+            cm_reward_min_ep, cm_reward_max_ep = self._compute_min_max_reward(self.total_nb_attacks,self.nb_last_attacks)
+            score_ep = self._normalisation_fun(self.cumulated_reward, cm_reward_min_ep, cm_reward_max_ep,self.min_score,self.max_score,self.blackout_encountered)
+
+            # TODO
+            #if self.blackout_encountered:
+            #    lines_attacked_dangerzone = (env._time_since_last_attack >= 0) * (env._time_since_last_attack < SURVIVOR_TIMESTEPS)
+            #
+            #    self.alert_attack_blackout += 1. * any(lines_alerted_beforeattack[lines_attacked_dangerzone])
+            #    self.no_alert_attack_blackout += 1. * any(~lines_alerted_beforeattack[lines_attacked_dangerzone])
+            #else :
+            #    lines_attacked_no_blackout = env._time_since_last_attack > 0
+            #
+            #    self.alert_attack_no_blackout += np.sum(lines_alerted_beforeattack[lines_attacked_no_blackout])
+            #    self.no_alert_attack_no_blackout += np.sum(~lines_alerted_beforeattack[lines_attacked_no_blackout])
+            
+            return score_ep
+        
+    @staticmethod
+    def _normalisation_fun(cm_reward, cm_reward_min_ep, cm_reward_max_ep,min_score,max_score,is_blackout):
+
+        #in case cm_reward_min_ep=cm_reward_max_ep=0, score can be maximum or 0
+        if cm_reward_min_ep == cm_reward_max_ep:
+            if is_blackout:
+                score_ep = 0.0
+            else:
+                score_ep = max_score
+        else:
+            standardized_score = (cm_reward - cm_reward_min_ep) / (cm_reward_max_ep - cm_reward_min_ep)#denominator cannot be 0 given first if condition
+            score_ep = min_score + (max_score - min_score) * standardized_score
+        return score_ep
+    
+    def _compute_min_max_reward(self, nb_attacks,nb_last_attacks):
+
+        if self.blackout_encountered:
+            if nb_last_attacks == 0: #in the case the blackout is not tied to a recent attack
+                cm_reward_min_ep = self.reward_min_no_blackout * nb_attacks
+                cm_reward_max_ep = self.reward_max_no_blackout * nb_attacks
+            elif nb_last_attacks >= 1:#in the case one or several recent attacks can be tied to the blackout 
```

Comment:
blackout is caused by the attack, not the opposite (rephrase if possible)

Reply:
`#in the case the blackout  can be tied to one or several recent attacks`

Reply:
Ok I did that